cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -263,10 +265,12 @@ where
                     })?,
             };
 
-            return Ok(Some(TransactionAndReceipt {
-                transaction,
-                receipt,
-            }));
+            if receipt.status == Some(TRANSACTION_STATUS_OK.into()) {
```

Comment:
Would this maybe be better to do in the `matches` function?

This fn here is more about composing existing functionality instead of acting on this level of detail.

Reply:
I hesitated. Happy to move it.

Reply:
Up to you, we have the feature now, can always improve that later :)